File: rpi/src/models/IMU.py
from pydantic import BaseModel, Field


class IMUData(BaseModel):
    """
    Represents the data from the IMU (Inertial Measurement Unit).
    """
    acceleration_x: float = Field(..., description="Acceleration in the X direction in m/s^2")
    acceleration_y: float = Field(..., description="Acceleration in the Y direction in m/s^2")
    acceleration_z: float = Field(..., description="Acceleration in the Z direction in m/s^2")
    gyroscope_x: float = Field(..., description="Angular velocity around the X axis in rad/s")
    gyroscope_y: float = Field(..., description="Angular velocity around the Y axis in rad/s")
    gyroscope_z: float = Field(..., description="Angular velocity around the Z axis in rad/s")
    temperature: float = Field(..., description="Temperature in degrees Celsius")

File: frontend/src/lib/types/IMU.ts
import { z } from 'zod';

export const IMUSchema = z.object({
    acceleration_x: z.number(),
    acceleration_y: z.number(),
    acceleration_z: z.number(),
    gyroscope_x: z.number(),
    gyroscope_y: z.number(),
    gyroscope_z: z.number(),
    temperature: z.number(),
});

export type IMU = z.infer<typeof IMUSchema>;